src/analytics/lambdas/scan-metadata-workflow/store-metadata-into-ddb.ts

```diff
@@ -0,0 +1,242 @@
+
+const REDSHIFT_DATA_API_ROLE_ARN = process.env.REDSHIFT_DATA_API_ROLE!;
+const REDSHIFT_DATABASE = process.env.REDSHIFT_DATABASE!;
+
+const { ddb_region, ddb_table_name } = parseDynamoDBTableARN(process.env.METADATA_DDB_TABLE_ARN!);
+
+// Create an Amazon service client object.
+const redshiftDataApiClient = getRedshiftClient(REDSHIFT_DATA_API_ROLE_ARN);
+
+export interface StoreMetadataEvent {
+  detail: StoreMetadataBody;
+}
+
+// Create an Amazon DynamoDB service client object.
+const ddbClient = new DynamoDBClient({
+  ...aws_sdk_client_common_config,
+  region: ddb_region,
+});
+
+/**
+ * The lambda function to get event metadata and properties metadata from redshift. And store them into DDB
+ * @param event StoreMetadataEvent.
+ * @returns.
+ */
+export const handler = async (event: StoreMetadataEvent) => {
+  logger.debug('request event:', JSON.stringify(event));
+
+  const metadataItems: any[] = [];
+  const appId = event.detail.appId;
+  try {
+    await handleEventMetadata(appId, metadataItems);
+
+    await handlePropertiesMetadata(appId, metadataItems);
+
+    await batchWriteIntoDDB(metadataItems);
+
+    return {
+      detail: {
+        appId: appId,
+        status: 'SUCCEED',
+        message: 'store metadata into ddb successfully',
+      },
+    };
+  } catch (err) {
+    if (err instanceof Error) {
+      logger.error('Error when query metadata.', err);
+    }
+    return {
+      detail: {
+        appId: appId,
+        status: 'FAILED',
+        message: 'store metadata into ddb failed',
+      },
+    };
+  }
+};
+
+async function handleEventMetadata(appId: string, metadataItems: any[]) {
+
+  const inputSql = `SELECT id, type, prefix, project_id, app_id, event_name, metadata_source, data_volumel_last_day, platform FROM ${appId}.event_metadata;`;
+
+  const response = await queryMetadata(inputSql);
+
+  // Transform data to DynamoDB format
+  const items = response.Records!.map(record => ({
+    PutRequest: {
+      Item: {
+        id: { S: record[0].stringValue },
+        type: { S: record[1].stringValue },
+        prefix: { S: record[2].stringValue },
+        projectId: { S: record[3].stringValue },
+        appId: { S: record[4].stringValue },
+        name: { S: record[5].stringValue },
+        displayName: { S: '' },
+        description: { S: '' },
+        hasData: { BOOL: true },
+        metadataSource: { S: record[6].stringValue },
+        dataVolumeLastDay: { N: record[7].stringValue },
+        platform: { L: convertToDDBList(record[8].stringValue) },
+        updateAt: { N: Date.now().toString() },
+        createAt: { N: Date.now().toString() },
+        operator: { S: '' },
+        deleted: { BOOL: false },
+        ttl: { N: (Math.floor(Date.now() / 1000) + 7 * 24 * 60 * 60).toString() },
+      },
+    },
+  }));
+
+  items.forEach(item => {
+    metadataItems.push({
+      PutRequest: {
+        Item: item.PutRequest.Item,
```

Comment:
Will each request only have one item?

Reply:
Yes, already refactored the code with lib-dynamodb change.